services/wallet/cryptocompare.go

```diff
@@ -127,3 +145,57 @@ func fetchTokenMarketValues(symbols []string, currency string) (map[string]Marke
 	return item, nil
 
 }
+
+func fetchHourlyMarketValues(symbol string, currency string, limit int, aggregate int) ([]TokenHistoricalPairs, error) {
+	item := []TokenHistoricalPairs{}
+	httpClient := http.Client{Timeout: time.Minute}
+
+	url := fmt.Sprintf("https://min-api.cryptocompare.com/data/v2/histohour?fsym=%s&tsym=%s&aggregate=%d&limit=%d&extraParams=Status.im", symbol, currency, aggregate, limit)
+	resp, err := httpClient.Get(url)
+	if err != nil {
+		return item, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return item, err
+	}
+
+	container := HistoricalValuesData{}
+	err = json.Unmarshal(body, &container)
+	if err != nil {
+		return item, err
+	}
+
+	item = container.Data.HistoricalData
+
+	return item, nil
+}
+
+func fetchDailyMarketValues(symbol string, currency string, limit int, allData bool, aggregate int) ([]TokenHistoricalPairs, error) {
+	item := []TokenHistoricalPairs{}
+	httpClient := http.Client{Timeout: time.Minute}
+
+	url := fmt.Sprintf("https://min-api.cryptocompare.com/data/v2/histoday?fsym=%s&tsym=%s&aggregate=%d&limit=%d&allData=%v&extraParams=Status.im", symbol, currency, aggregate, limit, allData)
```

Comment:
This base URL is used in multiple places now. Probably makes sense to move that into a reusable thing.

Reply:
done